rfcs/text/0007_lifecycle_unblocked.md

```diff
@@ -0,0 +1,339 @@
+- Start Date: 2019-09-11
+- RFC PR: (leave this empty)
+- Kibana Issue: (leave this empty)
+
+# Summary
+
+Prevent plugin lifecycle methods from blocking Kibana startup by making the
+following changes:
+1. Synchronous lifecycle methods 
+2. Synchronous context provider functions
+3. Core should not expose API's as observables
+
+# Motivation
+Plugin lifecycle methods and context provider functions are async
+(promise-returning) functions. Core runs these functions in series and waits
+for each plugin's lifecycle/context provider function to resolve before
+calling the next. This allows plugins to depend on the API's returned from
+other plugins.
+
+With the current design, a single lifecycle method or context provider that
+blocks will block all of Kibana from starting up. Plugins (including legacy
```

Comment:
This is not quite true, async context providers do not block Kibana start up. They could delay some routes from responding, but only for plugins that depend on the plugin that registered the context provider.

Reply:
Ah yes, I should split lifecycle methods and context providers cause the async impact is slightly different.